go/proxyd/backend.go

```diff
@@ -199,51 +306,219 @@ func (b *Backend) doForward(rpcReq *RPCReq) ([]byte, error) {
 
 type BackendGroup struct {
 	Name     string
-	backends []*Backend
-	i        int64
+	Backends []*Backend
 }
 
 func (b *BackendGroup) Forward(rpcReq *RPCReq) (*RPCRes, error) {
-	var outRes *RPCRes
-	for _, back := range b.backends {
+	rpcRequestsTotal.Inc()
+
```

Comment:
The iteration order for the backends is non deterministic, so is it possible for a single user to send 2 requests and have them go to different backends? Same with a user who is both using websockets and JSON RPC from the same app - an example would be subscribing to the tip and then querying for transaction data on each new tip

Reply:
Backend iteration order is deterministic. The backend would have to be marked offline for the case you are describing to happen. Essentially:

- User is connected via WS. Backend goes down, but stays connected via WS.
- User makes request via RPC. Request gets routed to the next online backend.
- Alternatively, a user makes two requests and the backend goes down in between requests 1 and 2.